src/Integration/Binding/BindingController.cs

```diff
@@ -95,12 +99,13 @@ protected override int OnQueryStatus(ref Guid pguidCmdGroup, uint cCmds, OLECMD[
         private bool OnBindStatus(BindCommandArgs args)
         {
             return args != null
-                && args.ProjectKey != null
-                && this.host.VisualStateManager.IsConnected
-                && !this.host.VisualStateManager.IsBusy
-                && VsShellUtils.IsSolutionExistsAndFullyLoaded()
-                && VsShellUtils.IsSolutionExistsAndNotBuildingAndNotDebugging()
-                && (this.projectSystemHelper.GetSolutionProjects()?.Any() ?? false);
+                   && args.ProjectKey != null
+                   && this.host.VisualStateManager.IsConnected
+                   && !this.host.VisualStateManager.IsBusy
+                   && (folderWorkspaceService.IsFolderWorkspace()
+                       || (VsShellUtils.IsSolutionExistsAndFullyLoaded()
+                           && VsShellUtils.IsSolutionExistsAndNotBuildingAndNotDebugging()
+                           && (this.projectSystemHelper.GetSolutionProjects()?.Any() ?? false)));
```

Comment:
`GetSolutionProjects` currently calls through to `ProjectToLanguageMapper.GetLanguageForProject`, so this condition will only be true if there is a bindable project.

We probably still want to check that condition for CMake projects, since it is a more strict check than just that the current solution is an "open as folder" one i.e.

```
...
&& (folderWorkspaceService.IsFolderWorkspace()
                       || (VsShellUtils.IsSolutionExistsAndFullyLoaded()
                           && VsShellUtils.IsSolutionExistsAndNotBuildingAndNotDebugging())
&& (this.projectSystemHelper.GetSolutionProjects()?.Any() ?? false)
```

Reply:
Yes you're right, this would allow us to easy disable it for JS 👍